xcarchive/xcarchive.go

```diff
@@ -39,3 +42,38 @@ func IsMacOS(archPath string) (bool, error) {
 
 	return exist, nil
 }
+
+func unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, fileName string) (string, error) {
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("__xcarhive__")
+	if err != nil {
+		return "", err
+	}
+
+	if err := ziputil.UnZip(xcarchivePth, tmpDir); err != nil {
+		return "", err
+	}
+
+	applicationsPth := filepath.Join(tmpDir, filepath.Base(xcarchivePth), "Products", "Applications")
+	appName := strings.TrimSuffix(filepath.Base(xcarchivePth), filepath.Ext(xcarchivePth))
+
+	return utility.FindFileInAppDir(applicationsPth, appName, fileName)
+}
+
+// UnwrapEmbeddedMobileProvision ...
+func UnwrapEmbeddedMobileProvision(xcarchivePth string) (string, error) {
+	return unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, "embedded.mobileprovision")
```

Comment:
I think these wrappers are not really needed, what about having the unwrapFileEmbeddedInXcarchiveDir exported only?

Reply:
Wrappers usually increase readability, reusability and are less error prone.